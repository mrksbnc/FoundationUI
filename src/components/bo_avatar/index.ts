export * from './bo_avatar.contant';
export * from './bo_avatar.type';
export { default as BoAvatar } from './BoAvatar.vue';
