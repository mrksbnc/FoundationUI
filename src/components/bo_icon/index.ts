export * from './bo_icon.constant';
export * from './bo_icon.type';
export { default as BoIcon } from './BoIcon.vue';
