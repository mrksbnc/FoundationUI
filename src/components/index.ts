export * from '@/components/bo_badge';
export * from '@/components/bo_button';
export * from '@/components/bo_icon';
export * from '@/components/bo_spinner';
