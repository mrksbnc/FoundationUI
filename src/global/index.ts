export * from './html_button';
export * from './html_input';
