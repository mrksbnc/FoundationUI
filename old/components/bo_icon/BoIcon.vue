<template>
	<!-- eslint-disable vue/no-v-html -->
	<div
		type="image/svg+xml"
		:style="style"
		class="bo-icon"
		:class="classes"
		aria-label="icon"
		v-html="svg"
	></div>
</template>

<script lang="ts">
export default defineComponent({
	inheritAttrs: true,
});
</script>

<script setup lang="ts">
import { IconSizeClasses, icons, type Icon } from './bo_icon';
import { BoSize } from '@/global';
import {
	ref,
	toRefs,
	watch,
	type PropType,
	defineComponent,
	computed,
	type StyleValue,
} from 'vue';

const props = defineProps({
	icon: {
		type: String as PropType<Icon>,
		required: true,
	},
	size: {
		type: String as PropType<BoSize>,
		default: () => BoSize.default,
	},
	color: {
		type: String,
		default: () => 'currentColor',
	},
});

const { icon, size, color } = toRefs(props);

const svg = ref('');

const iconMap = Object.keys(icons).reduce(
	(acc, key) => {
		const splitted = key.split('/');
		const icon = splitted[splitted.length - 1].split('.')[0];

		acc[icon] = icons[key];

		return acc;
	},
	{} as Record<string, () => Promise<string>>,
);

const style = computed<StyleValue>(() => {
	return {
		color: color.value,
	};
});

const classes = computed<string>(() => {
	return IconSizeClasses[size.value];
});

const load = async (): Promise<void> => {
	try {
		iconMap[icon.value]().then((val) => {
			svg.value = val;
		});
	} catch (e) {
		console.error(`Could not find icon of name ${icon.value}`);
	}
};

watch(icon, () => load(), { immediate: true });
</script>
