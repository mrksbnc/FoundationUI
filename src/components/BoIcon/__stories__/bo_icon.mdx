import { Canvas, Meta, Controls } from '@storybook/blocks';
import * as IconStories from './bo_icon.stories';

<Meta of={IconStories} />

# Icon

An icon is a small visual representation of an object or action. It's typically used to represent a button, a menu, or a status.

## Props

- `icon`: the name of the icon

<hr />

- `size`: the size of the icon

<hr />

- `color`: the color of the icon in `hex` format

<Canvas of={IconStories.Default} />
<Controls />

## Sizes

The icon can be displayed in different sizes. The default size is `default`, and the other sizes are `small`, `large`, and `extra_large`.

<Canvas of={IconStories.Sizes} />

## Colors

The icon can be displayed in different colors. The default color is `currentColor`, and the other colors are `black`, `white`, `red`, `green`, and `blue`.

<Canvas of={IconStories.Colors} />
