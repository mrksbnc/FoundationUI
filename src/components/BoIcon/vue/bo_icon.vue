<template>
	<div
		v-html="svg"
		:style="style"
		class="bo-icon"
		:class="classes"
		aria-label="icon"
		type="image/svg+xml"
	/>
</template>

<script lang="ts">
export default defineComponent({
	inheritAttrs: true,
});
</script>

<script setup lang="ts">
import { useIcon, Icon } from '@/components/BoIcon';
import { BoSize } from '@/constants';
import {
	ref,
	toRefs,
	watch,
	type PropType,
	defineComponent,
	computed,
	type StyleValue,
} from 'vue';
import { useIconStyle } from '../ts/bo_icon';

const props = defineProps({
	icon: {
		type: String as PropType<Icon>,
		required: true,
	},
	size: {
		type: String as PropType<BoSize>,
		default: () => BoSize.default,
	},
	color: {
		type: String,
		default: () => 'currentColor',
	},
});

const { icon } = toRefs(props);

const svg = ref('');

const style = computed<StyleValue>(() => {
	return useIconStyle(toRefs(props)).style;
});

const classes = computed<string>(() => {
	return useIconStyle(toRefs(props)).classes;
});

watch(
	() => icon.value,
	async () => {
		svg.value = await useIcon(toRefs(props)).getIconSvg(icon.value);
	},
	{ immediate: true },
);
</script>
