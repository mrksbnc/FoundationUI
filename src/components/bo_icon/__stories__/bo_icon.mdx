import { Canvas, Meta, Controls } from '@storybook/blocks';
import * as BoIconStories from './bo_icon.stories';

<Meta
	title="Components/bo-icon"
	component={BoIconStories.BoIcon}
/>

# bo-icon

An icon is a graphical representation of an object or action. The `bo-icon` component is a component that can be used to display icons in various ways.
This library uses the [Feather Icons](https://feathericons.com/) collection of icons.

## Props

- `icon` - The name of the icon (required)
  - an icon name from the `Icon` enum
- `size` - The size of the icon (optional)
  - default: `BoSize.default`
- `color` - The color of the icon in `hex` format (optional)
  - default: `'currentColor'`

## Examples

<Canvas of={BoIconStories.Example} />
<Controls of={BoIconStories.Example} />

## Sizes

The `size` prop can be used to change the size of the icon. The `size` prop should be a member of the `BoSize` enum.

- `extra_small`
- `small`
- `default`
- `large`
- `extra_large`

<Canvas of={BoIconStories.Sizes} />

## Colors

The `color` prop can be used to change the color of the icon. The `color` prop should be a string that represents the color in `hex` format.
The `color` prop can take the values of the `BoColor` enum or any other color value that can be used until it's in valid `hex` format.

<Canvas of={BoIconStories.Colors} />

## List of all icons

<Canvas of={BoIconStories.AllIcons} />
