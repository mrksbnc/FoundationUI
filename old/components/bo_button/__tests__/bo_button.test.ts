import { Icon } from '@/components/bo_icon';
import { BoSize } from '@/global';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';
import {
	BoButton,
	BoButtonBorderRadiusClasses,
	BoButtonFilledColorClasses,
	BoButtonType,
	BoButtonVariant,
} from '..';

let wrapper: ReturnType<typeof mount>;

beforeEach(() => {
	wrapper = mount(BoButton, {
		props: {
			label: 'Label',
			variant: BoButtonVariant.primary,
			type: BoButtonType.default,
			disabled: false,
			size: BoSize.default,
		},
	});
});

describe('bo_button.vue', () => {
	test('should match snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	test('should render the button', () => {
		expect(wrapper.find('button')).toBeTruthy();
	});

	test('should have the default size', () => {
		expect(wrapper.find('w-6')).toBeTruthy();
		expect(wrapper.find('h-6')).toBeTruthy();
	});

	test('should have the default type', () => {
		expect(wrapper.find('bg-blue-500')).toBeTruthy();
		expect(wrapper.find('rounded-lg')).toBeTruthy();
	});

	test('should be able to change the size', async () => {
		await wrapper.setProps({ size: BoSize.large });
		expect(wrapper.find('w-8')).toBeTruthy();
		expect(wrapper.find('h-8')).toBeTruthy();
	});

	test('should be able to change the type', async () => {
		await wrapper.setProps({ type: BoSize.large });
		expect(wrapper.find('bg-blue-500')).toBeTruthy();
		expect(wrapper.find('rounded-lg')).toBeTruthy();
	});

	test('should be able to change the color', async () => {
		await wrapper.setProps({ color: '#000000' });
		expect(wrapper.find('bg-blue-500')).toBeTruthy();
		expect(wrapper.find('rounded-lg')).toBeTruthy();
	});

	test('should be able to change the variant', async () => {
		await wrapper.setProps({ variant: BoButtonVariant.secondary });
		expect(wrapper.find('bg-gray-600')).toBeTruthy();
		expect(wrapper.find('rounded-lg')).toBeTruthy();
	});

	test('should be able to change the type', async () => {
		await wrapper.setProps({ type: BoButtonType.pill });
		expect(wrapper.find('bg-blue-600')).toBeTruthy();
		expect(wrapper.find('rounded-full')).toBeTruthy();
	});

	test('should be able to change the disabled', async () => {
		await wrapper.setProps({ disabled: true });
		expect(wrapper.find('disabled')).toBeTruthy();
	});

	test('should be able to change the prefix icon', async () => {
		await wrapper.setProps({ prefixIcon: Icon.alert_circle });
		expect(wrapper.find('svg')).toBeTruthy();
	});

	test('should be able to change the suffix icon', async () => {
		await wrapper.setProps({ suffixIcon: Icon.alert_circle });
		expect(wrapper.find('svg')).toBeTruthy();
	});

	test('should be able to change the border radius', async () => {
		await wrapper.setProps({ borderRadius: BoButtonBorderRadiusClasses.pill });
		expect(wrapper.find('rounded-full')).toBeTruthy();
	});

	test('should be able to change the fill color', async () => {
		await wrapper.setProps({ fillColor: BoButtonFilledColorClasses.primary });
		expect(wrapper.classes().join(' ')).includes(
			'bg-blue-600 hover:bg-blue-700 focus:ring-blue-600 text-white',
		);
	});
});
