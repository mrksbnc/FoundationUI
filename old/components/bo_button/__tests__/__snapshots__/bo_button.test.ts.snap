// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bo_button.vue > should match snapshot 1`] = `
"<button type="button" role="button" class="bg-blue-600 hover:bg-blue-700 focus:ring-blue-600 text-white shadow-sm shadow-blue-500/50 dark:shadow-sm dark:shadow-blue-800/80 px-3 py-1.5 flex gap-1 items-center justify-center transition-opacity duration-200 cursor-pointer shadow-md disabled:cursor-not-allowed disabled:opacity-50 text-sm rounded-lg">
  <!--v-if--><span class="bo-button__label">Label</span>
  <!--v-if-->
</button>"
`;
